data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeCoercionServiceTest.java

```diff
@@ -0,0 +1,141 @@
+        assertThat(result, instanceOf(Float.class));
+        assertThat(result, equalTo(testFloat));
+    }
+
+    @Test
+    void testCoerceTerminalNodeJsonPointerType() throws ParseTreeCoercionException {
+        final String testKey1 = "key1";
+        final String testKey2 = "key2";
+        final String testValue = "value";
+        final String testJsonPointerKey = String.format("/%s/%s", testKey1, testKey2);
+        final Event testEvent = createTestEvent(Map.of(testKey1, Map.of(testKey2, testValue)));
+        when(token.getType()).thenReturn(DataPrepperExpressionParser.JsonPointer);
+        when(terminalNode.getText()).thenReturn(testJsonPointerKey);
+        final Object result = objectUnderTest.coerceTerminalNode(terminalNode, testEvent);
+        assertThat(result, instanceOf(String.class));
+        assertThat(result, equalTo(testValue));
+    }
+
+    @Test
+    void testCoerceTerminalNodeJsonPointerTypeMissingKey() throws ParseTreeCoercionException {
+        final String testMissingKey = "missingKey";
+        final Event testEvent = createTestEvent(new HashMap<>());
+        when(token.getType()).thenReturn(DataPrepperExpressionParser.JsonPointer);
+        when(terminalNode.getText()).thenReturn(testMissingKey);
+        final Object result = objectUnderTest.coerceTerminalNode(terminalNode, testEvent);
+        assertThat(result, nullValue());
+    }
+
+    @Test
+    void testCoerceTerminalNodeEscapedJsonPointerType() throws ParseTreeCoercionException {
+        final String testKey = "key";
+        final String testValue = "value";
+        final String testEscapedJsonPointerKey = String.format("\"/%s\"", testKey);
+        final Event testEvent = createTestEvent(Map.of(testKey, testValue));
+        when(token.getType()).thenReturn(DataPrepperExpressionParser.EscapedJsonPointer);
+        when(terminalNode.getText()).thenReturn(testEscapedJsonPointerKey);
+        final Object result = objectUnderTest.coerceTerminalNode(terminalNode, testEvent);
+        assertThat(result, instanceOf(String.class));
+        assertThat(result, equalTo(testEscapedJsonPointerKey));
+    }
+
+    @Test
+    void testCoerceTerminalNodeEscapedUnsupportedType() {
+        final Event testEvent = createTestEvent(new HashMap<>());
+        when(token.getType()).thenReturn(-1);
+        assertThrows(ParseTreeCoercionException.class, () -> objectUnderTest.coerceTerminalNode(terminalNode, testEvent));
+    }
+
+    @Test
+    void testCoerceSuccess() throws ParseTreeCoercionException {
+        final Object testObj = false;
+        final Boolean result = objectUnderTest.coerce(testObj, Boolean.class);
+        assertThat(result, instanceOf(Boolean.class));
+        assertThat(result, is(false));
+    }
+
+    @Test
+    void testCoerceFailure() {
+        final Object testObj = false;
+        assertThrows(ParseTreeCoercionException.class, () -> objectUnderTest.coerce(testObj, String.class));
```

Comment:
This test might be more readable if you provided a `mock(Object.class)` instead of `false`. Presently, it seems that `false` means something specific, but you are just creating a different type.

Reply:
Unfortunately one can not mock getClass method since it is final. I will use TestObject here to indicate the difference.